src/domain/event/occurrences/OccurrencesTable.tsx

```diff
@@ -314,6 +314,7 @@ const OccurrenceEnrolmentTable: React.FC<{
 
   return (
     <Table
+      // @ts-ignore
```

Comment:
Why was this added? [Typescript documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments) says:
> It is recommended practice to have the remainder of the comment following @ts-ignore explain which error is being suppressed.

Reply:
Adding `"noErrorTruncation": true` to `compilerOptions` in tsconfig.json makes it possible to see the entire error message without @ts-ignore:

`> npx tsc --project ./tsconfig.json --noEmit`:

```typescript
src/domain/event/occurrences/OccurrencesTable.tsx:317:7 - error TS2322: Type '({ id: string; Header: string; 'aria-hidden': boolean; accessor?: undefined; Cell?: undefined; style?: undefined; } | { accessor: (row: OccurrenceFieldsFragment) => string | null; id: string; Header?: undefined; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => Element | "–"; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => Element | "-"; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => string; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { accessor: (row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment; Cell: ({ row, value, }: { value: OccurrenceFieldsFragment; row: { getToggleRowExpandedProps: () => ButtonProps & RefAttributes<HTMLButtonElement>; isExpanded: boolean; }; }) => Element; id: string; style: { width: string; }; Header?: undefined; 'aria-hidden'?: undefined; })[]' is not assignable to type 'Column<OccurrenceFieldsFragment>[]'.
  Type '{ id: string; Header: string; 'aria-hidden': boolean; accessor?: undefined; Cell?: undefined; style?: undefined; } | { accessor: (row: OccurrenceFieldsFragment) => string | null; id: string; Header?: undefined; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => Element | "–"; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => Element | "-"; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { Header: string; accessor: (row: OccurrenceFieldsFragment) => string; id: string; 'aria-hidden'?: undefined; Cell?: undefined; style?: undefined; } | { accessor: (row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment; Cell: ({ row, value, }: { value: OccurrenceFieldsFragment; row: { getToggleRowExpandedProps: () => ButtonProps & RefAttributes<HTMLButtonElement>; isExpanded: boolean; }; }) => Element; id: string; style: { width: string; }; Header?: undefined; 'aria-hidden'?: undefined; }' is not assignable to type 'Column<OccurrenceFieldsFragment>'.
    Type '{ accessor: (row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment; Cell: ({ row, value, }: { value: OccurrenceFieldsFragment; row: {    getToggleRowExpandedProps: () => React.ComponentProps<typeof Button>;    isExpanded: boolean;}; }) => JSX.Element; id: string; style: { width: string; }; Header?: undefined; 'aria-hidden'?: undefined; }' is not assignable to type 'Column<OccurrenceFieldsFragment>'.
      Type '{ accessor: (row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment; Cell: ({ row, value, }: { value: OccurrenceFieldsFragment; row: {    getToggleRowExpandedProps: () => React.ComponentProps<typeof Button>;    isExpanded: boolean;}; }) => JSX.Element; id: string; style: { width: string; }; Header?: undefined; 'aria-hidden'?: undefined; }' is not assignable to type 'ColumnInterface<OccurrenceFieldsFragment> & { accessor: "seatsTaken"; } & ColumnInterfaceBasedOnValue<OccurrenceFieldsFragment, number>'.
        Type '{ accessor: (row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment; Cell: ({ row, value, }: { value: OccurrenceFieldsFragment; row: {    getToggleRowExpandedProps: () => React.ComponentProps<typeof Button>;    isExpanded: boolean;}; }) => JSX.Element; id: string; style: { width: string; }; Header?: undefined; 'aria-hidden'?: undefined; }' is not assignable to type '{ accessor: "seatsTaken"; }'.
          Types of property 'accessor' are incompatible.
            Type '(row: OccurrenceFieldsFragment) => OccurrenceFieldsFragment' is not assignable to type '"seatsTaken"'.

317       columns={columns}
          ~~~~~~~

  src/common/components/table/Table.tsx:9:3
    9   columns: Array<Column<D>>;
        ~~~~~~~
    The expected type comes from property 'columns' which is declared here on type 'IntrinsicAttributes & Props<OccurrenceFieldsFragment>'


Found 1 error in src/domain/event/occurrences/OccurrencesTable.tsx:317
```

Reply:
I investigated it for a while earlier. It might be that the 3rd party TableComponent that we used there does not support React 18 properly. The feature itself seems to work, so I just added the `@ts-ignore` and didn't want to waste more time with that at this point. But if you are eager to find a fix for that and find one, that's a good thing! :)

Reply:
@nikomakela Hopefully "yarn typecheck" passes for you locally as well now, I'm not sure what happened, the typescript error above disappeared at some point and didn't reappear, weird! Hopefully the problem is gone now!

Reply:
Tried cloning the palvelutarjotin-ui repository afresh, ran "yarn install" and "yarn typecheck" and the typecheck went through without any warnings/errors so it should be fine I think.